### DISCUSSION TITLE
implicitly has type 'any' because it does not have a type annotation

### DISCUSSION THREAD

**Discussion by productdevbook** [colinhacks/zod#1657] [Q&A] [answered]
```ts
const LocationSchema = z.object({
  aa: z.object({
    title: z.string(),
  }),
  bb: z.object({
    type: z.literal('Point'),
    coordinates: z.array(z.number()),
  }),
  school: SchoolSchema,
})
```

```ts
const SchoolSchema = z.object({
  title: z.string(),
  address: z.string(),
  locations: z.array(LocationSchema),
})
```

'LocationSchema' implicitly has type 'any' because it does not have a type annotation and is referenced directly or indirectly in its own initializer.ts(7022)

**Comment by JacobWeisenburger** [comment, marked as answer]
These might be helpful for you:
[Recursive types](https://github.com/colinhacks/zod#recursive-types)
[Cyclical objects](https://github.com/colinhacks/zod#cyclical-objects)

**Reply by santosh-akqa** [reply]
Yes but when we extend schema with [Recursive types](https://github.com/colinhacks/zod#recursive-types), i.e.


```
const baseCategorySchema = z.object({
  name: z.string(),
});

type Category = z.infer<typeof baseCategorySchema> & {
  subcategories: Category[];
};

const categorySchema: z.ZodType<Category> = baseCategorySchema.extend({
  subcategories: z.lazy(() => categorySchema.array()),
});
```

like

`const productSchema = categorySchema.extend(....)`

we'll get
> Property 'extend' does not exist on type 'ZodType<BaseContainer, ZodTypeDef, BaseContainer>'

@JacobWeisenburger Can you suggest any solution for this 🧐
